bot/exts/fun/duck_pond.py

```diff
@@ -165,7 +169,9 @@ async def on_raw_reaction_add(self, payload: RawReactionActionEvent) -> None:
         if not self._payload_has_duckpond_emoji(payload.emoji):
             return
 
-        channel = discord.utils.get(self.bot.get_all_channels(), id=payload.channel_id)
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(payload.guild_id)
+        channel = discord.utils.get(guild.text_channels + guild.threads, id=payload.channel_id)
```

Comment:
I wasn't aware that this was a thing.

Although looking at the docs, that would also get voice channels etc., not just TextChannel or Thread. Is this something that matters?

Reply:
Well, I wouldn't worry about getting reactions inside a voice channel.

Reply:
Surely it would break stuff. E.g. on L183 we do `channel.fetch_message`, which would raise an error for a VoiceChannel (voice channels don't have such a method)

Reply:
I'll update this to use `Guild.get_channel_or_thread`